### ISSUE TITLE
Unable to detect `None` in nested `Literal`

### ISSUE THREAD

**Issue by dhruvmanila** [microsoft/pyright#5221]
**Describe the bug**

Pyright is complaining when `None` is used in nested `Literal` when the error message clearly says that type arguments for "Literal" can be None.

**To Reproduce**

Use the mentioned code below and run `pyright` on it.

**Expected behavior**

This should be allowed.

**Screenshots or Code**

```python
from typing import Literal

# `None` in nested literal doesn't work
# 
# error: Type arguments for "Literal" must be None, a literal value (int, bool, str, or bytes), or an enum value
foo: Literal[10, Literal["foo", None]]

# This is fine
bar: Literal[10, Literal["foo", "bar"], None]
```

**VS Code extension or command-line**

* Running from command-line
* Version: `pyright 1.1.311`

**Additional context**

`mypy` doesn't complain about this and using `reveal_type` it unpacks the nested `Literal` correctly:

```python
from typing import Literal

foo: Literal[10, Literal["foo", None]]
reveal_type(foo)
# note: Revealed type is "Union[Literal[10], Literal['foo'], None]"
```

**Comment by erictraut** [comment]
Thanks for the bug report. This will be fixed in the next release.

**Comment by dhruvmanila** [comment]
Awesome, thanks for the quick fix! :)

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.312, which I just published. It will also be included in the next insiders release of pylance.
